tests/test_settings.py

```diff
@@ -60,28 +60,42 @@ def test_change_username(client: FlaskClient, user: User) -> None:
 
 @pytest.mark.usefixtures("_authenticated_user")
 def test_change_password(client: FlaskClient, user: User, user_password: str) -> None:
-    new_password = user_password + "xxx"
-
     assert len(original_password_hash := user.password_hash) > 32
     assert original_password_hash.startswith("$scrypt$")
     assert user_password not in original_password_hash
 
-    response = client.post(
-        url_for("settings.change_password"),
-        data={
-            "old_password": user_password,
-            "new_password": new_password,
-        },
-        follow_redirects=True,
-    )
-    assert response.status_code == 200
-    assert "Password successfully changed. Please log in again." in response.text
-    assert "/login" in response.request.url
-    assert len(new_password_hash := user.password_hash) > 32
-    assert new_password_hash.startswith("$scrypt$")
-    assert original_password_hash not in new_password_hash
-    assert user_password not in new_password_hash
-    assert new_password not in new_password_hash
+    for new_password in [user_password, "", "aB!!", "aB3!", (33 * "aB3!")[:129], 5 * "aB3!"]:
```

Comment:
we should use pytest's `parametrize` mark to inject these if possible. it makes it easier to see which of the cases fail

Reply:
There's two issues with going the parametrize route:

1. Because we still haven't solved the slow test problem, using parametrize can add about a second or more of execution time to the tests for each piece of example data. That's a bit painful.
2. The test could use a refactor, because it's structured to test everything in one big flow. I'd like to have several tests with declarative names, like ``test_empty_new_password_entry_is_invalid``, etc. But because of the slow tests, this better pattern slows down the tests dramatically. This should probably wait until we can speed up the tests.

It's a little verbose, but for now I can add context data to the end of each assert, so it's shown if that line throws.

Reply:
Addressed by (https://github.com/scidsg/hushline/pull/635/commits/7761f1d6a0b68421bfd918928615b639e6373746)